specification/servicebus/resource-manager/Microsoft.ServiceBus/2017-04-01/servicebus.json

```diff
@@ -3103,6 +3223,236 @@
+		"x-ms-client-flatten": true,
+			"properties": {
+				"code": {
+				  "readOnly": true,
+				  "type": "string",
+				  "description": "Region code"
+				},
+				"fullName": {
+				  "readOnly": true,
+				  "type": "string",
+				  "description": "Full name of the region"
+				}
+			}
+		}
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/RegionResource"
+        }
+      ],
+      "description": "Premium Messaging Region"
+    },
+    "PremiumMessagingRegionsListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PremiumMessagingRegions"
+          },
+          "description": "Result of the List PremiumMessagingRegions type."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Link to the next set of results. Not empty if Value contains incomplete list of PremiumMessagingRegions."
+        }
+      },
+      "description": "The response of the List PremiumMessagingRegions operation."
+    },
+    "Eventhub": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "properties": {
+            "partitionIds": {
+              "readOnly": true,
+              "type": "array",
+              "items": {
+                "type": "string"
+              },
+              "description": "Current number of shards on the Event Hub."
+            },
+            "createdAt": {
+              "readOnly": true,
+              "format": "date-time",
+              "type": "string",
+              "description": "Exact time the Event Hub was created."
+            },
+            "updatedAt": {
+              "readOnly": true,
+              "format": "date-time",
+              "type": "string",
+              "description": "The exact time the message was updated."
+            },
+            "messageRetentionInDays": {
+              "format": "int64",
+              "type": "integer",
+              "maximum": 7,
+              "minimum": 1,
+              "description": "Number of days to retain the events for this Event Hub, value should be 1 to 7 days"
+            },
+            "partitionCount": {
+              "format": "int64",
+              "type": "integer",
+              "maximum": 32,
+              "minimum": 1,
+              "description": "Number of partitions created for the Event Hub, allowed values are from 1 to 32 partitions."
+            },
+            "status": {
+              "type": "string",
+              "description": "Enumerates the possible values for the status of the Event Hub.",
+              "enum": [
+                "Active",
+                "Disabled",
+                "Restoring",
+                "SendDisabled",
+                "ReceiveDisabled",
+                "Creating",
+                "Deleting",
+                "Renaming",
+                "Unknown"
+              ],
+              "x-ms-enum": {
```

Comment:
Setting ```modelAsString``` to false means any changes to the enum would be a breaking change, please reconsider this.